jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarStreamBuilder.java

```diff
@@ -61,6 +68,17 @@ public void addEntry(TarArchiveEntry entry) {
     entries.add(entry);
   }
 
+  /** Adds a blob to the archive. */
+  public void addEntry(Blob blob, String name) throws IOException {
+    // TODO: Efficiency? We're writing blob contents twice (once here to get entry size, once when
```

Comment:
Yeah, I think you could do it by creating a class that encompasses either a file or a blob and a name, and it has a method to write it to a TarArchiveOutputStream with the right TarArchiveEntry and contents.